HIRS_Utils/src/main/java/hirs/tpm/eventlog/uefi/UefiDevicePath.java

```diff
@@ -0,0 +1,441 @@
+  byte[] fileName = convertChar16tobyteArray(filePath);
+  tmpType += new String(fileName, "UTF-8");
+  subType = tmpType;
+  return tmpType;
+}
+
+/**
+ * Process a vendor sub-type on a Media Type.
+ * Length of this structure in bytes. Length is 20 + n bytes
+ * Vendor-assigned GUID that defines the data that follows.
+ * Vendor-defined variable size data.
+ * @param path
+ * @param offset
+ * @return vendor device info.
+ */
+private String vendorSubType(final byte[] path, final int offset) {
+  String tmpType = "Vendor Subtype GUID = ";
+  byte[] lengthBytes = new byte[UefiConstants.SIZE_2];
+  System.arraycopy(path, UefiConstants.OFFSET_2 + offset, lengthBytes, 0, UefiConstants.SIZE_2);
+  int subTypeLength = HexUtils.leReverseInt(lengthBytes);
+  byte[] guidData = new byte[UefiConstants.SIZE_16];
+  System.arraycopy(path, UefiConstants.OFFSET_4 + offset, guidData, 0, UefiConstants.SIZE_16);
+  UefiGuid guid = new UefiGuid(guidData);
+  tmpType += guid.toString() + " ";
+  if (subTypeLength - UefiConstants.SIZE_16 > 0) {
+    byte[] vendorData = new byte[subTypeLength - UefiConstants.SIZE_16];
+    System.arraycopy(path, UefiConstants.OFFSET_20
+            + offset, vendorData, 0, subTypeLength - UefiConstants.SIZE_16);
+    tmpType += " : Vendor Data = " + HexUtils.byteArrayToHexString(vendorData);
+  } else {
+    tmpType += " : No Vendor Data pesent";
+  }
+    subType = tmpType;
+    return tmpType;
+}
+
+/**
+ * Returns nvm device info.
+ * UEFI Specification, Version 2.8.
+ * Name space Identifier (NSID) and IEEE Extended Unique Identifier (EUI-64):
+ * See Links to UEFI Related Documents
+ * (http://uefi.org/uefi under the headings NVM Express Specification.
+ * @param path
+ * @param offset
+ * @return NVM device info.
+ */
+private String nvmSubType(final byte[] path, final int offset) {
+  String tmpType = "NVM Express Namespace = ";
+  byte[] lengthBytes = new byte[UefiConstants.SIZE_2];
+  System.arraycopy(path, UefiConstants.OFFSET_2 + offset, lengthBytes, 0, UefiConstants.SIZE_2);
+  int subTypeLength = HexUtils.leReverseInt(lengthBytes);
+  byte[] nvmData = new byte[subTypeLength];
+  System.arraycopy(path, UefiConstants.OFFSET_4 + offset, nvmData, 0, subTypeLength);
+  tmpType += HexUtils.byteArrayToHexString(nvmData);
+  subType = tmpType;
+  return tmpType;
+}
+
+/**
+ * BIOS Device Type definition.
+ * From Appendix A of the BIOS Boot Specification.
+ * Only process the Device type.
+ * Status bootHandler pointer, and description String pointer are ignored.
+ * @param path byte array holding the device path.
+ * @return String that represents the UEFI defined BIOS Device Type.
+ */
+private String biosDevicePath(final byte[] path, final int offset) {
+  String devPath = "Legacy BIOS : Type = ";
+  byte devPathType = path[offset + 1];
+  Byte pathType = Byte.valueOf(devPathType);
+  switch (pathType.intValue()) {
+    case UefiConstants.DEVPATH_BIOS_RESERVED: devPath += "Reserved"; break;
```

Comment:
This case is a duplicate of the default statement, either the default statement should change or this case should be moved right above the default and the "break" removed.

Reply:
Changed default to "Unknown".